docs/json-codec-backend/codec-docs.md

```diff
@@ -0,0 +1,97 @@
+## JSON serialization support for generated Scala classes
+
+The purpose of this documentation is to give an explanation of how the JSON Codec Backend works and how to use it.
+This backend is a feature built on top of the Scala backend which allows you to read Scala types that the Scala backend generates from JSON and write to JSON.
+
+### How to use the JSON Codec Backend
+
+The first step to using the JSON Codec backend is to add the Circe JSON library and ```morphir sdk``` as dependencies to your project. The ```morphir sdk``` can be added to your project through the following ways:
+
+1. Copy dependencies from ```morphir-jvm``` : This option allows you to copy the dependencies(morphir sdk) used by the generated code to the output path. You can achieve this by adding the ```-c``` or ```--copy-deps``` flag to the ```morphir-gen``` command.
+
+2. Adding from `Maven Repository` : The SDK can be added from the maven repository as a dependency . It can be found at [Morphir SDK Core](https://mvnrepository.com/artifact/org.morphir/morphir-sdk-core).
+
+
+#### Generating Codecs from the IR
+The ```morphir-ir.json``` is a prerequisite to generating codecs. Make sure you have that and proceed with the following steps
+
+1.  Run ```morphir-elm gen --include-codecs --target Scala``` to generate the codecs
+2. In the situation where you do not have ```morphir sdk``` as a dependency in your project, you have to add the ```--copy-deps``` flag to copy dependencies from the ```morphir-jvm``` project. Complete command for this is ```morphir-elm gen --iclude-codecs --copy-deps --target Scala```.
+
+### How to use Generated Codecs
+Using an example , lets describe a model structure and look at how the generated codec for this particular model will be structured
+
+The model:
+
+	- reference-model
+      - src
+        - Morphir
+          - Reference
+	        - Model
+              - BooksAndRecords.elm  
+	- morphir.JSON
+	- morphir-ir.JSON
```

Comment:
I think you did search and replace and now JSON is upper-case here too where it shouldn't be.